app/assets/javascripts/discourse/app/components/topic-timeline/scroll-area.js

```diff
@@ -0,0 +1,231 @@
+import GlimmerComponent from "discourse/components/glimmer";
+import { bind } from "discourse-common/utils/decorators";
+import { tracked } from "@glimmer/tracking";
+import discourseLater from "discourse-common/lib/later";
+import { action } from "@ember/object";
+
+export const SCROLLER_HEIGHT = 50;
+const MIN_SCROLLAREA_HEIGHT = 170;
+const MAX_SCROLLAREA_HEIGHT = 300;
+const LAST_READ_HEIGHT = 20;
+
+export default class TopicTimelineScrollArea extends GlimmerComponent {
+  @tracked showButton = false;
+  @tracked scrollPosition;
+  @tracked current;
+  @tracked percentage = this._percentFor(
+    this.args.topic,
+    this.args.enteredIndex + 1
+  );
+  @tracked total;
+  @tracked date;
+  @tracked lastReadPercentage = null;
+  @tracked position;
+  @tracked lastReadTop = Math.round(
+    this.lastReadPercentage * scrollareaHeight()
+  );
+
+  style = `height: ${scrollareaHeight()}px`;
+  before = this.scrollareaRemaining() * this.percentage;
+  after = scrollareaHeight() - this.before - SCROLLER_HEIGHT;
+
+  get showDockedButton() {
+    return !this.site.mobileView && this.hasBackPosition && !this.showButton;
+  }
+
+  get hasBackPosition() {
+    return (
+      this.lastRead &&
+      this.lastRead > 3 &&
+      this.lastRead > this.current &&
+      Math.abs(this.lastRead - this.current) > 3 &&
+      Math.abs(this.lastRead - this.total) > 1 &&
+      this.lastRead !== this.total
+    );
+  }
+
+  get beforePadding() {
+    return `height: ${this.before}px`;
+  }
+
+  get afterPadding() {
+    return `height: ${this.after}px`;
+  }
+
+  get lastReadStyle() {
+    return `height: ${LAST_READ_HEIGHT}px; top: ${this.topPosition}px`;
+  }
```

Comment:
These are triggering the Ember 'binding style attributes' warning in the console. We'll need to `htmlSafe()` them per https://deprecations.emberjs.com/v1.x/#toc_binding-style-attributes